pkg/cmd/cli/cmd/newapp.go

```diff
@@ -532,6 +532,10 @@ func CompleteAppConfig(config *newcmd.AppConfig, f *clientcmd.Factory, c *cobra.
 	if config.BinaryBuild && config.Strategy == generate.StrategyPipeline {
 		return kcmdutil.UsageError(c, "specifying binary builds and the pipeline strategy at the same time is not allowed.")
 	}
+
+	if len(config.BuildArgs) > 0 && config.Strategy != generate.StrategyDocker {
+		return kcmdutil.UsageError(c, "build arguments can only be used with Docker builds.")
+	}
```

Comment:
i have mixed feelings as it is a slippery slope, but yes i guess if someone explicitly indicated they want to use build-args, we should throw an error if we aren't going to produce a docker build object.